src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/MethodBodyScanner.cs

```diff
@@ -123,8 +123,8 @@ private static Stack<StackSlot> MergeStack(Stack<StackSlot> a, Stack<StackSlot>
             }
 
             Stack<StackSlot> newStack = new Stack<StackSlot>(a.Count);
-            IEnumerator<StackSlot> aEnum = a.GetEnumerator();
-            IEnumerator<StackSlot> bEnum = b.GetEnumerator();
+            var aEnum = a.GetEnumerator();
```

Comment:
This is breaking our repo rules about the use of `var`:

https://github.com/dotnet/runtime/blob/2f63046c28f2f57f2c8666845f1c1de8696b1626/docs/coding-guidelines/coding-style.md#L22

I'd honestly prefer suppressing CA1859 in the ILCompiler.Compiler, ILCompiler.TypeSystem projects (project-wide) because I don't see a good change in this diff.

Reply:
Ok, used concrete type instead of var (to keep the code in sync with illink) and reverted the rest of the changes and suppressed the warning.